dts: arm: st: Factorize dts files on F1, F3 and L4 series

erwango opened the pull request:
As it is done on F4 family, introduce a stm32xx.dtsi file in each stm32 series.
This new file is the smallest common denominator for each series.
Besides, introduce the notion of inheritance between SoCs (for instance stm32f103xe is 
a superset of stm32103xb and hence it is now reflected in dtsi files by inclusion)

This allows to reduce global size of stm32 dtsi files and hence minimize maintenance efforts.

PS: Inhenritance between socs could be seen by using diff on cmsis soc files 
(ext/hal/st/stm32cube/stm32<yy>xx/soc/*.h files)

erwango committed:
dts: arm: st: Factorize STM32F1 series soc dtsi files

In order to simplify maintenance of dts files for
stm32f1 series, introduce a stm32f1.dtsi file which
represent the smallest common denominator of IPs in
the family.
Besides, stm32f103Xe includes stm32f103xb, as it is a
extension of this SoC.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
dts: arm: st: Factorize STM32F3 series soc dtsi files

In order to simplify maintenance of dts files for
stm32f1 series, introduce a stm32f3.dtsi file which
represent the smallest common denominator of IPs in
the family.
Besides, stm32f334 includes stm32f303, as it is a
extension of this SoC.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ydamigos commented:
I totally agree with the introduction of a stm32xx.dtsi file which is the smallest common denominator for each series,  but we should do it for the SoCs in the family not only the ones which are now supported by Zephyr. For example in F1, I2C2 device is not supported by all SoCs (e.g. stm32f103x6), so the i2c2 node should be moved to the stm32103xb dtsi file.

ydamigos commented on dts/arm/st/stm32f1.dtsi:
```diff
@@ -0,0 +1,83 @@
+			label = "UART_2";
+		};
+
+		usart3: serial@40004800 {
```
usart3 should be moved to the stm32103xb dtsi file

erwango replied:
cf comment on defining "minimal platform"

ydamigos commented on dts/arm/st/stm32f1.dtsi:
```diff
@@ -0,0 +1,83 @@
+			label= "I2C_1";
+		};
+
+		i2c2: i2c@40005800 {
```
i2c2 should be moved to the stm32103xb dtsi file

erwango replied:
cf comment on defining "minimal platform"

erwango commented:
@ydamigos: This was indeed my initial idea (valid also for L4 and F3).
Though, this require work (today and in everyday maintenance) for socs that might never be ported on zephyr.
Aim here was to reduce effective maintenance cost to the minimum.
This will indeed require additional work and changes when introducing stm32f103x6 (or stm32f10X..), but  
this is hypothetical.

ydamigos commented:
@erwango If the plan is to reduce effective maintenance cost to the minimum and make the additional changes when we introduce a new SoC, it's ok for me. It will make the introduction of the new dtsi file a little more confusing, but we could handle it when the time comes.

erwango committed:
dts: arm: st: Factorize STM32L4 series soc dtsi files

In order to simplify maintenance of dts files for
stm32f4 series, introduce a stm32l4.dtsi file which
represent the smallest common denominator of IPs in
the family.
This allows to fix usart4 availability on stm32l432
which was not correct.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@galak, I see 2 cons on this:
-Some IPs are not compatible with each others in the same series (different address, different properties). How to address this?
-Each time you add a new node in the master file, you need to take care to delete it in all the soc.dtsi that don't include it. I see it hard to maintain

galak commented:
I was mostly saying the delete node might be useful for some cases of standard IP but where the numbers vary (like uart's, i2c, etc).  So the base soc would be minimal plus

erwango commented:
ok, I'm not too much in favor of that (and I know STM32 Linux side neither), but maybe we can come back on that once there will be more socs available. We'll have a better view at this time.
